samples/star/src/androidTest/kotlin/com/slack/circuit/star/petdetail/PetDetailTest.kt

```diff
@@ -127,4 +132,37 @@ class PetDetailTest {
       }
     }
   }
+
+  @Test
+  fun petDetail_emits_event_when_tapping_on_full_bio_button() = runTest {
+    val channel = Channel<Any>(1)
+
+    val success =
+      PetDetailScreen.State.Success(
+        url = "url",
+        photoUrls = listOf("http://some.url"),
+        photoUrlMemoryCacheKey = null,
+        name = "Baxter",
+        description = "Grumpy looking Australian Terrier",
+        tags = listOf("dog", "terrier", "male"),
+        eventSink = channel::trySend
+      )
+
+    val circuitConfig =
+      CircuitConfig.Builder()
+        .setOnUnavailableContent { screen ->
+          PetPhotoCarousel(PetPhotoCarouselScreen.State(screen as PetPhotoCarouselScreen))
+        }
+        .build()
+
+    composeTestRule.run {
+      setContent { CircuitCompositionLocals(circuitConfig) { PetDetail(success) } }
+
+      onNodeWithTag(CAROUSEL_TAG).assertIsDisplayed().performTouchInput { swipeUp() }
+      onNodeWithTag(FULL_BIO_TAG, true).assertIsDisplayed().performClick()
+
+      val event = channel.receive()
+      assertThat(event).isEqualTo(PetDetailScreen.Event.ViewFullBio(success.url))
+    }
```

Comment:
Out of curiosity why do we do these run blocks?

```suggestion
    composeTestRule.setContent {
      setContent { CircuitCompositionLocals(circuitConfig) { PetDetail(success) } }
    }

    onNodeWithTag(CAROUSEL_TAG).assertIsDisplayed().performTouchInput { swipeUp() }
    onNodeWithTag(FULL_BIO_TAG, true).assertIsDisplayed().performClick()

    val event = channel.receive()
    assertThat(event).isEqualTo(PetDetailScreen.Event.ViewFullBio(success.url))
```

Reply:
Same [comment](https://github.com/slackhq/circuit/pull/275/files/9978fe2c84c0299c9ff14359acfe4bde665dee91#r1023324483) here